### PR TITLE
FEC-10464 - send youbora ad start event when on the correct time

### PR DESCRIPTION
**Pull request by giladna**
and not in ad loaded event

### CODE REVIEW

File: youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java [hunk 1]
```diff
@@ -225,8 +226,10 @@ private void addListeners() {
             if (isNullAdapter()) {
                 return;
             }
-            getPlugin().getAdapter().fireStart();
-            fireStart();
+            if (currentAdInfo != null && currentAdInfo.getAdPositionType() == AdPositionType.PRE_ROLL) {
+                getPlugin().getAdapter().fireStart();
+                fireStart();
+            }
```
Comment (OrenMe) [inline youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java]: why do we need to distinguish pre and mid/post?

**Review by OrenMe — CHANGES_REQUESTED** [review]
The correct behaviour is to fireStart on adStarted and fire join on first progress event of ad.

File: youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java (hunk 1 above)
Comment (giladna) [reply youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java]: the issue is only in mid rolls where the IMA preloading so event is coming too early
so kept the original code for pre-rolls which is ok
and fixed the issue which exists only in mid-rolls and post-rolls

moving the for fire  (of ad-adapter and youbora-adapter) code to other places broke the events flow

Comment (OrenMe) [reply youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java]: as the adStarted is correct for all ad types keeping this code is redundant and will just get us code duplicates to maintain. 
Also, the ad join is incorrect now as it is being fired with ad start - and this has always been a mistake.
Move it to ad progress.
We don't even have to add protection to it as youbora lib does so you can call it each time for a given ad as youbora library will persist the state until the ad has ended.

See js implementation:
on ad start:
https://github.com/kaltura/playkit-js-youbora/blob/90f46c259f55514b42598b2a169d99c2823211df/src/adapter/ads/nativeads.js#L153-L158

on ad progress:
https://github.com/kaltura/playkit-js-youbora/blob/90f46c259f55514b42598b2a169d99c2823211df/src/adapter/ads/nativeads.js#L223-L226

also there is no need to add the code in ad buffered as IMA model will first fire started and only then buffered.

Comment (giladna) [reply youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java]: @OrenMe 
the join is ok the start will be in ad buffer start
this is only fallback to no such event fired

in the start position is already > 0

Comment (giladna) [reply youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java]: the thing is that the progress in ima is fired before start in IMA right now

**Comment by giladna** [comment]
@OrenMe

**Comment by giladna** [comment]
> The correct behaviour is to fireStart on adStarted and fire join on first progress event of ad.

In android this is bit different since the started event already position is > 0 so this is not accurate

**Review by OrenMe — APPROVED** [review]

**Pull request merged**